### PR TITLE
feat(ui): Implement Autocomplete and ConnectedAutocomplete

### PR DESCRIPTION
**Pull request by bdauria**

### CODE REVIEW

File: src/InputField/Select/Autocomplete/Connected/index.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      })
+      .catch(() => setOptions([]))
+      .finally(() => setLoading(false));
+  };
+
+  const changeText = (event): void => {
+    loadOptions(getSearchEndpoint(event.target.value));
+  };
+
+  const doOpen = (): void => {
```
[the remark this answers is not in the document]
Comment (bdauria) [reply src/InputField/Select/Autocomplete/Connected/index.tsx]: There is a variable named open already, so I had to chose another name for the function.

**Comment by bdauria** [comment]
@kduret There is no visual difference between the Autocomplete and the ConnectedAutocomplete, so in my opinion, the existing story covers it already. 

I might add a story for the loading cases though.

File: src/InputField/Select/Autocomplete/index.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        key={option.id}
+        label={option.name}
+        size="small"
+        {...getTagProps({ index })}
+      />
+    ));
+
+  return (
+    <Autocomplete
+      multiple
```
Comment (Thebarda) [inline src/InputField/Select/Autocomplete/index.tsx]: Maybe in another PR but Autocomplete could not allow multiple value

File: src/api/useCancelTokenSource.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+import axios, { CancelTokenSource } from 'axios';
+
+const useCancelTokenSource = (): CancelTokenSource => {
+  const [cancelTokenSource] = useState(axios.CancelToken.source());
+
+  return cancelTokenSource;
+};
+
+export default useCancelTokenSource;
```
Comment (Thebarda) [inline src/api/useCancelTokenSource.ts]: Is it the time to also export `useCancelTokenSource` from centreon-ui ?

**Comment by kduret** [comment]
> @kduret There is no visual difference between the Autocomplete and the ConnectedAutocomplete, so in my opinion, the existing story covers it already.
> 
> I might add a story for the loading cases though.

Ok no problem. But IMHO, the storybook should be also a developer guide which show some code examples, even if it renders the same visual

File: src/InputField/Select/Autocomplete/Connected/index.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  getSearchEndpoint: (searchField: string) => string;
+  getOptionsFromResult: (result) => Array<SelectEntry>;
+}
+
+const ConnectedAutocompleteField = <TData extends Record<string, unknown>>({
+  baseEndpoint,
+  getSearchEndpoint,
+  getOptionsFromResult,
+  ...props
+}: Props & AutocompleteFieldProps): JSX.Element => {
```
Comment (jdelpierre) [inline src/InputField/Select/Autocomplete/Connected/index.tsx]: I don't understand this syntax

Comment (bdauria) [reply src/InputField/Select/Autocomplete/Connected/index.tsx]: The ConnectedAutocompleteField is a 'generic' Component, with the type being expressed between the <>. If you look at the code below, you'll notive that the generic TData type is used to define the type of options received from the endpoint. More documentation here: https://www.typescriptlang.org/docs/handbook/generics.html

the Props & AutocompleteFieldProps is a union type between the declared Props type (declared above) and the actuel Autocomplete Props (declare as AutocompleteFieldProps). What we're doing here is basically augmenting the Prop type from the Autocomplete with 3 new props.

**Pull request merged**